timemachine/fe/bar.py

```diff
@@ -217,3 +263,52 @@ def pair_overlap_from_ukln(u_kln: np.ndarray) -> float:
     assert u_kn.shape == (k, l * n)
     N_k = n * np.ones(l)
     return 2 * pymbar.MBAR(u_kn, N_k).computeOverlap()["matrix"][0, 1]  # type: ignore
+
+
+def compute_fwd_and_reverse_df_over_time(
+    ukln_by_lambda: NDArray, chunks: int = 10
+) -> Tuple[NDArray, NDArray, NDArray, NDArray]:
+    """Provided a u_kln computes the forward and reverse dF estimates.
+
+    Computes the dF value for increasing numbers of samples in both the forward and reverse direction.
+
+    Parameters
+    ----------
+
+    ukln_by_lambda: [n_lambda, 2, 2, N] array
+        Array of u_kln broken up by lambda windows
+
+    chunks: int
+        Number of chunks to break up u_kln into
+
+    Returns
+    -------
+        fwd_df: [chunks] np.ndarray
+            numpy array of dF for each chunk
+        fwd_df_err: [chunks] np.ndarray
+            numpy array of dF errors for each chunk
+        rev_df: [chunks] np.ndarray
+            numpy array of dF for each chunk, from reversed ukln
+        rev_df_err: [chunks] np.ndarray
+            numpy array of dF errors for each chunk, from reversed ukln
+    """
+    assert len(ukln_by_lambda.shape) == 4
+    assert ukln_by_lambda.shape[1] == 2
+    forward_predictions_ = []
+    reverse_predictions_ = []
+    total_frames = ukln_by_lambda.shape[-1]
+    assert total_frames >= chunks, "fewer samples than chunks"
+    frame_step_size = total_frames // chunks
```

Comment:
Expose this instead of chunks? It might be more intuitive to think in terms of frame step size vs chunks.